src/main/java/ua/tc/marketplace/jwtAuth/JwtUtil.java

```diff
@@ -0,0 +1,81 @@
+package ua.tc.marketplace.jwtAuth;
+
+import io.jsonwebtoken.*;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+import ua.tc.marketplace.model.entity.User;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@AllArgsConstructor
+public class JwtUtil {
+    private final JwtConfig jwtConfig;
+
+    public String createToken(User user) {
+        Claims claims = Jwts.claims().setSubject(user.getEmail());
+        claims.put("firstName",user.getFirstName());
+        claims.put("lastName",user.getLastName());
+        Date tokenCreateTime = new Date();
+        Date tokenValidity = new Date(tokenCreateTime.getTime()
+                + TimeUnit.MINUTES.toMillis(jwtConfig.getTokenExpirationAfterDays()*24*60));
+        return Jwts.builder()
+                .setClaims(claims)
+                .setExpiration(tokenValidity)
+                .signWith(SignatureAlgorithm.HS256, jwtConfig.getSecretKey())
+                .compact();
+    }
+
+    private Claims parseJwtClaims(String token) {
+        return Jwts.parser()
+                .setSigningKey(jwtConfig.getSecretKey())
+                .parseClaimsJws(token).getBody();
+    }
+
+    public Claims resolveClaims(HttpServletRequest req) {
+        try {
+            String token = resolveToken(req);
+            if (token != null) {
+                return parseJwtClaims(token);
+            }
+            return null;
+        } catch (ExpiredJwtException ex) {
+            req.setAttribute("expired", ex.getMessage());
```

Comment:
need to understand how is this working. And make it work as we need. We need to send to Client correct error message of our standard form. And print into app's console only one line with `ERROR` tag and actual error message